sdk/src/main/java/com/amplitude/experiment/storage/Cache.kt

```diff
@@ -0,0 +1,66 @@
+package com.amplitude.experiment.storage
+
+import com.amplitude.experiment.util.toMap
+import org.json.JSONObject
+
+internal class LoadStoreCache<V>(
+    private val namespace: String,
+    private val storage: Storage,
+    private val transformer: ((value: Any) -> V)? = null
```

Comment:
Transformer should be required here, since we have to transform the unstructured json to a strongly typed object. (with javascript we can just parse and cast the json.